lasagne/layers/base.py

```diff
@@ -300,3 +307,50 @@ def get_output_for(self, inputs, **kwargs):
         `NotImplementedError`.
         """
         raise NotImplementedError
+
+
+class UnLayer(MergeLayer):
+
+    """
+    lasagne.layers.UnLayer(incoming, layer, **kwargs)
+
+    The :class:`UnLayer` class performing inverse operations
+    for a single layer of a neural network: transposed layer
+    for a :class:`DenseLayer`, deconvolutional layer for
+    :class:`Conv2DLayer`, :class:`Conv1DLayer`; or
+    an unpooling layer for :class:`MaxPool2DLayer`.
+
+    It is specially useful for building (convolutional)
+    autoencoders with tied parameters.
```

Comment:
I'd add something like: "Note that if the layer to be inverted contains a nonlinearity and/or a bias, the :class:`UnLayer` will include the derivative of that in its computation. If you want to avoid that, each encoder layer should have its nonlinearities and biases moved into a :class:`NonlinearityLayer`." (That will require adding a bias to `NonlinearityLayer`.)